ecosystem/sep-0043.md

```diff
@@ -0,0 +1,146 @@
+simplifying the onboarding of new wallets.
+
+## Specification
+
+The standard web wallet interface provides a structure for common methods that wallets are expected to offer. In
+addition, the interface defines common error message shapes to ensure wallets adhere to informative and usable error
+messaging. Messages are designed to be surfaced directly to the user, so they should provide a readable description of
+what went wrong.
+
+### Wallet Interface Format
+
+Wallets will employ an API using the following format:
+
+```typescript
+
+interface Error {
+  message: string,    // general description message returned to the client app
+  code: number,       // unique error code
+  ext?: Array<string>  // optional extended details
+}
+
+{
+  getAddress: () => Promise<{address: string;} | Error>,
+
+  signTransaction: (xdr: string, opts?: { networkPassphrase?: string, address?: string; submit?: boolean; submitUrl?: string; }) => Promise<{signedTxXdr: string; signerAddress: string;} | Error>,
+
+  signAuthEntry: (authEntry: string, opts?: { networkPassphrase?: string, address?: string }) => Promise<{signedAuthEntry: string; signerAddress: string;} | Error>,
+
+  signMessage: (message: string, opts?: { networkPassphrase?: string, address?: string }) => Promise<{signedMessage: string; signerAddress: string;} | Error>,
+
+  getNetwork: () => Promise<{network: string, networkPassphrase: string}>
+}
+```
+
+In addition, this proposal has a standard for error interfaces. Below are examples of different kinds of errors a wallet
+may surface:
+
+```typescript
+{
+  unhandled: { // internal wallet error, likely caused by the wallet logic itself
+    message: 'The wallet encountered an internal error. Please try again or contact the wallet if the problem persists.',
+    code: -1
+  },
+  external: { // external service (Horizon, RPC, etc.) returned an error
+    message: 'An error occurred with Horizon. Please try again.',
+    code: -2,
+    ext: ['Operation 2: Invalid "amount" value.', "Operation 5: Asset issuer is required."] // malformed tx error example
+  },
+  invalid: { // client app request is invalid (wrong parameters, invalid transaction XDR, etc.)
+    message: 'Request is invalid. Please check the details and try again.',
+    code: -3,
+    ext: ['Invalid transaction XDR'] // example error
+  },
+  rejected: { // user rejected the request, client app should not try to retry the request
+    message: 'The user rejected this request.',
+    code: -4
+  }
+}
+
+```
+
+### Methods
+
+### getAddress
+
+`getAddress` will provide the public key the wallet is signing for. This may be the account loaded in the wallet, or it
+may be a different account that the wallet is the signer for.
+
+### signTransaction
+
+`signTransaction` will take an XDR provided by the requester and apply a signature to it before returning a signed XDR
+and the signer address back to the requester. The first parameter is the XDR string while the second parameter is a set
```

Comment:
Suggested rephrase to provide more detail, somewhat applies to everywhere you mention XDR strings

```suggestion
`signTransaction` will take a serialized transaction provided by the requester and apply a signature to it before returning a signed XDR and the signer address back to the requester. The first parameter is a base64-encoded string of an XDR `Transaction` instance (right? or is it the envelope? plus cross-reference to the `.x` file/line itself) while the second parameter is a set
```

Also, does

> signed XDR

mean a `TransactionEnvelope` with a corresponding `DecoratedSignature` appended?